### PR TITLE
Show json instead of tables in dev mode and show summaries

### PR DESCRIPTION
**Pull request by Henry-Pulver**
Shows the short summary in non-dev mode, full text in dev mode

https://github.com/Superflows-AI/chat-ui/assets/33871096/358d62b2-db7a-49f1-9d08-04ae43e127a4

### CODE REVIEW

File: src/components/SuperflowsSidebar.tsx [hunk 1]
```diff
@@ -370,18 +376,10 @@ export default function SuperflowsSidebar(props: {
...
                         }
                         return (
-                          <DevChatItem
-                            chatItem={{
-                              ...chatItem,
-                              content: contentString,
-                            }}
+                          <UserChatItem
+                            chatItem={{ ...chatItem, content: contentString }}
                             key={idx.toString() + chatItem.content}
```
Comment (neuromantic99) [inline src/components/SuperflowsSidebar.tsx:383]: same comment as the superflows repo

File: src/components/chatItems.tsx [hunk 2]
```diff
@@ -53,17 +53,9 @@ export function DevChatItem(props: {
...
-    graphedData = extractGraphData(props.chatItem.content);
     try {
-      const functionJsonResponse = JSON.parse(props.chatItem.content) as Json;
-      if (functionJsonResponse && typeof functionJsonResponse === "object") {
-        content = convertToRenderable(
-          functionJsonResponse,
-          `${functionNameToDisplay(props.chatItem?.name ?? "")} result`,
-        );
-      }
+      content = JSON.stringify(JSON.parse(props.chatItem.content), null, 2);
```
Comment (neuromantic99) [inline src/components/chatItems.tsx:58]: can this error?

File: src/components/chatItems.tsx [hunk 3]
```diff
@@ -239,6 +229,26 @@ export function UserChatItem(props: {
...
+          functionJsonResponse,
+          `${functionNameToDisplay(props.chatItem?.name ?? "")} result`,
+        );
+      }
+    } catch {
+      // If not JSON, then there may be a summary
+      content = props.chatItem.summary ?? props.chatItem.content;
+    }
+  }
+  const [tabOpen, setTabOpen] = useState<"table" | "graph">("table");
```
Comment (neuromantic99) [inline src/components/chatItems.tsx:250]: This logic is a bit confusing for me, you switch to the summary if the attempt to JSON the content errors?

Comment (Henry-Pulver) [reply src/components/chatItems.tsx:250]: Yep, we only summarise non-JSON responses, so if it fails to convert to JSON, there may be a summary.

If it's JSON, there's definitely no summary, so no point in checking

File: src/components/chatItems.tsx (hunk 2 above)
Comment (Henry-Pulver) [reply src/components/chatItems.tsx:58]: Yes, hence it's inside a try-catch

**Pull request merged**